python_modules/dagster/dagster/_grpc/impl.py

```diff
@@ -86,6 +86,14 @@ def core_execute_run(
 
     # try to load the pipeline definition early
     try:
+        # add in cached metadata to load repository more efficiently
+        if pipeline_run.has_repository_metadata:
```

Comment:
skeptical on the stored bool - do we end up fetching this a plan anyway down the call stack?

Reply:
In cases where there we have solids_to_execute or asset_selection, we don't end up fetching the stored plan, and instead create a new one in this process (https://sourcegraph.com/github.com/dagster-io/dagster/-/blob/python_modules/dagster/dagster/_core/execution/api.py?L952:5)

So in cases where we have an asset selection and no repository load metadata, this check would save us from having to pull that from the instance. Sympathetic to the argument that this is not a particularly important optimization, so happy to take it out.